### PR TITLE
OAK-10367: update org.apache.sling.testing.osgi-mock dependency

### PR DESCRIPTION
**Pull request by mbaedke**

### CODE REVIEW

File: oak-auth-external/pom.xml [hunk 1]
```diff
@@ -69,6 +69,16 @@
             <artifactId>org.osgi.service.component</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```
Comment (kwin) [inline oak-auth-external/pom.xml]: which code changes requires this new compile classpath change?

File: oak-auth-external/pom.xml [hunk 2]
```diff
@@ -187,7 +197,27 @@
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
-            <artifactId>org.apache.sling.testing.osgi-mock</artifactId>
+            <artifactId>org.apache.sling.testing.osgi-mock.junit4</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (kwin) [inline oak-auth-external/pom.xml:193]: junit4 depends on core, therefore you only reference the former in the dependency usually in this case (https://sling.apache.org/documentation/development/osgi-mock.html#maven-dependency)

File: oak-auth-external/pom.xml [hunk 3]
```diff
@@ -187,7 +197,27 @@
...
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.testing.osgi-mock.core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (kwin) [inline oak-auth-external/pom.xml]: Why all the additional dependencies? Isn't the transitive dependencies from osgi-mock enough?
It is unusual to reference OSGi Spec chapter dependencies only in test classpath. In tests those are only used usually if the to-be tested code relies on those spec chapters as well.

File: oak-auth-external/pom.xml (hunk 1 above)
Comment (mbaedke) [reply oak-auth-external/pom.xml]: No code changes, these were listed as used undeclared dependencies by the dependency plugin.

File: oak-auth-external/pom.xml (hunk 2 above)
Comment (mbaedke) [reply oak-auth-external/pom.xml:193]: Again a used undeclared dependency, which I usually declare explicitly as generally recommended. In this particular case, it's admittedly unnecessary.

File: oak-auth-external/pom.xml (hunk 3 above)
Comment (mbaedke) [reply oak-auth-external/pom.xml]: The integration tests will fail without them.

Comment (kwin) [reply oak-auth-external/pom.xml]: but all those are transitive ones of https://github.com/apache/sling-org-apache-sling-testing-osgi-mock/blob/45fb63cb61615663e10ed3c9e612a3a3f6161c3e/core/pom.xml#L42 and the ITs don't leverage those APIs/spec chapter directly (only the Sling Testing Mock uses them).

**Comment by kwin** [comment]
I am not convinced, that we should make all those dependencies direct ones. Please only make those direct dependencies which are really used in Oak code and also please do that dependency cleanup in a separate commit to ease to distinguish it from the actual update of the testing dependency.

File: oak-auth-external/pom.xml (hunk 3 above)
Comment (mbaedke) [reply oak-auth-external/pom.xml]: Yeah. But the tests will fail anyway. Do you have a suggestion?
Maven dependency analysis does not see reflective use.

**Comment by mbaedke** [comment]
> I am not convinced, that we should make all those dependencies direct ones. Please only make those direct dependencies which are really used in Oak code 

If the dependency plugin lists them as used and undeclared, they are (in most cases) used in the project code, no?
In particular, org.osgi.service.component.annotations is used in oak-auth-external.
I don't have strong feelings about this, I just wonder. 

> and also please do that dependency cleanup in a separate commit to ease to distinguish it from the actual update of the testing dependency.

Agreed.

File: oak-segment-tar/src/test/java/org/apache/jackrabbit/oak/segment/SegmentBlobTrackerRegistrationTest.java [hunk 4]
```diff
@@ -19,36 +19,51 @@
...
-
     @Override
     protected void registerNodeStoreService() {
-        Map<String, Object> properties = newHashMap();
+        Hashtable<String, Object> properties = new Hashtable<>();
         properties.put(SegmentNodeStoreService.CUSTOM_BLOB_STORE, true);
         properties.put(SegmentNodeStoreService.REPOSITORY_HOME_DIRECTORY, repoHome);
-        service = context.registerInjectActivateService(new SegmentNodeStoreService(), properties);
+        SegmentNodeStoreService service = new SegmentNodeStoreService();
+
```
Comment (mbaedke) [inline oak-segment-tar/src/test/java/org/apache/jackrabbit/oak/segment/SegmentBlobTrackerRegistrationTest.java:48]: @kwin Would you take a look at the following lines? I wonder if MockBundleContext introduced a bug in the new version or if I'm missing something.

**Comment by reschke** [comment]
I'm getting test failures:

~~~
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 0.51 s <<< FAILURE! - in org.apache.jackrabbit.oak.segment.azure.Azu
reSegmentStoreServiceTest
[ERROR] deactivate(org.apache.jackrabbit.oak.segment.azure.AzureSegmentStoreServiceTest)  Time elapsed: 0.009 s  <<< ERROR!
java.lang.NoClassDefFoundError: org/osgi/service/event/EventAdmin
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at org.apache.sling.testing.mock.osgi.context.OsgiContextImpl.registerDefaultServices(OsgiContextImpl.java:68)
        at org.apache.sling.testing.mock.osgi.context.OsgiContextImpl.setUp(OsgiContextImpl.java:49)
        at org.apache.sling.testing.mock.osgi.junit.OsgiContext.access$100(OsgiContext.java:35)
        at org.apache.sling.testing.mock.osgi.junit.OsgiContext$1.before(OsgiContext.java:79)
        at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at com.arakelian.docker.junit.DockerRule$StatementWithDockerRule.evaluate(DockerRule.java:76)
        at org.apache.jackrabbit.oak.blob.cloud.azure.blobstorage.AzuriteDockerRule$1.evaluate(AzuriteDockerRule.java:112)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
~~~

**Review by reschke — CHANGES_REQUESTED** [review]
see https://github.com/apache/jackrabbit-oak/pull/1044/#issuecomment-1662386190

**Comment by mbaedke** [comment]
Yes, my Docker isn't working properly, which leads to test cases silently being skipped (obviously an issue of it's own). I'm working on it.

**Comment by mbaedke** [comment]
Integration test now succeed for me.

**Review by reschke — APPROVED** [review]

**Pull request merged**